src/main/java/com/gotocompany/depot/redis/util/RedisSinkUtils.java

```diff
@@ -27,4 +28,15 @@ public static Map<Long, ErrorInfo> getErrorsFromResponse(List<RedisRecord> redis
         );
         return errors;
     }
+
+    public static Map<Long, ErrorInfo> getNonRetryableErrors(List<RedisRecord> redisRecords, RedisNonRetryableException e, Instrumentation instrumentation) {
+        Map<Long, ErrorInfo> errors = new HashMap<>();
+        for (RedisRecord record : redisRecords) {
+            instrumentation.logError("Error while inserting to redis for message. Record: {}, Error: {}",
+                    record.toString(), e.getMessage());
+            errors.put(record.getIndex(), new ErrorInfo(new Exception(e.getMessage()), ErrorType.SINK_4XX_ERROR));
```

Comment:
Let's add one ErrorType.SINK_NON_RETRYABLE_ERROR 
this error type will be set in firehose for failing.